.github/actions/bpftrace/scripts/check-encryption-leaks.bt

```diff
@@ -390,9 +400,9 @@ kprobe:__dev_queue_xmit
   }
 
   if ($proto == PROTO_IPV4) {
-    @trace_ip4[$ip4h->saddr, $udph->source, $l4proto] = true;
+    @trace_ip4[$ip4h->saddr, $udph->source, $l4proto] |= SK_TRACK_FLAG_MASQUERADE;
   } else {
-    @trace_ip6[$ip6h->saddr.in6_u.u6_addr8, $udph->source, $l4proto] = true;
+    @trace_ip6[$ip6h->saddr.in6_u.u6_addr8, $udph->source, $l4proto] |= SK_TRACK_FLAG_MASQUERADE;
```

Comment:
Can you help me understand why packets reaching __dev_queue_xmit can be marked as `SK_TRACK_FLAG_MASQUERADE`? `__dev_queue_xmit` seems to be kernel function that has nothing to do with masquerading?

Reply:
The s-masq would happen before, between the packet creation (`kprobe:tcp_connect`) and its actual xmission through dev (`kprobe:__dev_queue_xmit`), in case we do need to perform s-masq.
Does this make sense? :thinking:

Reply:
what i don't understand is, by the time `br_forward` is triggered, the packets must have passed __dev_queue_xmit@eth0, no? This flag will always be set?

Reply:
Yep, this flag will always be set. However:

1. if the flow has been s-masq, then the value in the map will be equal to `SK_TRACK_FLAG_MASQUERADE`
2. if the flow has not been s-masq, then the value in the map will be equal to `SK_TRACK_FLAG_BASE | SK_TRACK_FLAG_MASQUERADE`

Maybe the naming is a bit counter-intuitive...